dat/board.lua

```diff
@@ -470,31 +500,86 @@ local function can_cannibalize ()
    return false
 end
 
-local function board_cannibalize ()
+local function can_cannibalize_usefully()
+   local pp = player.pilot()
+   local parmour, _pshield, _pstress = pp:health(true)
+   local ba=board_plt:stats().armour
+
+   if (pp:stats().armour == parmour) then
+      return false
+   elseif (ba<=2) or (ba==3 and player.shipvarPeek("cannibal2")) then
+      return false
+   else
+      return true
+   end
+end
+
+local function _board_cannibalize(spare)
    local armour, shield = board_plt:health(true)
    local cannibal2 = player.shipvarPeek("cannibal2")
 
-   if armour <= 1 then
-      return
-   end
    local bs = board_plt:stats()
 
    local pp = player.pilot()
    local ps = pp:stats()
    local parmour, pshield, pstress = pp:health(true)
 
-   local dmg = math.min( (armour-1), 2*(ps.armour-parmour) )
-   if dmg <= 0 then
-      return
+   local dmg
+
+   -- Because can't refresh board window, else would be useless.
+   spare=spare and can_cannibalize_usefully()
+
+   if not spare then
+      dmg = armour
+   else
+      -- Just take what you both need and can
+      if cannibal2 then
+         dmg = math.min( (armour-1), 3*(ps.armour-parmour)/2 )
+      else
+         dmg = math.min( (armour-1), 2*(ps.armour-parmour) )
+      end
    end
 
    board_plt:setHealth( 100*(armour-dmg)/bs.armour, 100*shield/bs.shield, 100 )
-   local heal_armour = dmg/2
+
+   local heal_armour
    if cannibal2 then
       heal_armour = dmg*2/3
+   else
+      heal_armour = dmg/2
+   end
+
+   if parmour+heal_armour>ps.armour then
+      heal_armour=ps.armour-parmour
+   end
+
+   if heal_armour > 0 then
+      pp:setHealth( 100*(parmour+heal_armour)/ps.armour, 100*pshield/ps.shield, pstress )
+      player.msg(fmt.f(_("Your ship cannibalized {armour} armour from {plt}."),{armour=fmt.number(heal_armour), plt=board_plt}))
+      board_fcthit_apply( heal_armour )
+   end
+
+   local left=armour-dmg
+
+   if left<=0 then
+      local fact=board_plt:faction()
+      player.msg(fmt.f(_("{plt} was destroyed."),{lft=fmt.number(left), plt=board_plt}))
+      fact:hit( -(board_plt:points()), system.cur(), "destroy")
+      board_close()
+   else
+      player.msg(fmt.f(_("{plt} was left with {lft} armour. Next time it won't !"),{lft=fmt.number(left), plt=board_plt}))
```

Comment:
Missing a verb no?
`Next time it won't survive!`
or
`Next time it will be destroyed!`